tests/basic/test-database.cpp

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2019 - 2022 Geode-solutions
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+#include <geode/basic/bitsery_archive.h>
+#include <geode/basic/database.h>
+#include <geode/basic/identifier.h>
+#include <geode/basic/logger.h>
+#include <geode/basic/uuid.h>
+
+#include <geode/tests/common.h>
+
+struct Foo : public geode::Identifier
+{
+    Foo() = default;
+    Foo( double value ) : value_( value ) {}
+
+    template < typename Archive >
+    void serialize( Archive& archive )
+    {
+        archive.ext( *this, bitsery::ext::BaseClass< geode::Identifier >{} );
+        archive.value8b( value_ );
+    }
+
+    double value_{ 0 };
+};
+
+void register_foo_serializer( geode::PContext& context )
+{
+    context
+        .registerSingleBaseBranch< geode::Serializer, geode::Identifier, Foo >(
+            "Foo" );
+}
+
+void register_foo_deserializer( geode::PContext& context )
+{
+    context.registerSingleBaseBranch< geode::Deserializer, geode::Identifier,
+        Foo >( "Foo" );
+}
```

Comment:
This is mandatory to (de)serialize directly from a smart pointer of an Identifier and let Bitsery saves/loads the right polymorphic type.